pkg/controllers/management/clusterprovisioner/helm_cleaner.go

```diff
@@ -0,0 +1,230 @@
+package clusterprovisioner
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/helm/helm-mapkubeapis/pkg/mapping"
+	"github.com/pkg/errors"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/mod/semver"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	apiMappings = &mapping.Metadata{
+		Mappings: []*mapping.Mapping{
+			{
+				DeprecatedAPI:    "apiVersion: policy/v1beta1\nkind: PodSecurityPolicy\n",
+				RemovedInVersion: "v1.25",
+				// TODO add all APIs we want to cover here. Perhaps store it in a ConfigMap so we don't need to change
```

Comment:
I'd guess we could go with the full set of https://github.com/helm/helm-mapkubeapis/blob/00164c79fa9bfc92827843bb694fcb64d1ef325c/config/Map.yaml, WDYT?

Reply:
IMO we should only pull in things for k8s 1.25 - I wouldn't want to risk this firing for other resources outside of the list for 1.25. 

However, we also need to make sure that removing these resources won't cause things to break. IMO the psp -> psa case is pretty simple, users would just need to add an annotation to a namespace, but if we end up doing something like removing a cronjob from a chart that could break something and not be too easy to recover from.

Reply:
That's fair. The reason why I didn't add other APIs here preemptively is because it would become more difficult to test. I'll review the deprecations doc for Kubernetes v1.25 and update the code in a separate commit accordingly. For now, just adding the Windows and ordering variations for PSP.

Reply:
I'm also hoping we can move this bit to something more configurable -- a ConfigMap, perhaps. I'd like some input/feedback on this idea though.

Reply:
I think the windows/line return stuff is more important. I do think that you are right that we'd want this to be a configmap or some such so that customers can modify/request a cleanup on each cluster on an as-needed basis. But that might be a wider conversation.

Reply:
That's fair. I can start out a document describing the idea for us to discuss at a later time.